src/controller/data_model/BUILD.gn

```diff
@@ -29,40 +30,142 @@ chip_data_model("data_model") {
 }
 
 if (current_os == "android") {
-  pw_python_action("java-jni-generate") {
-    script = "${chip_root}/scripts/codegen.py"
+  chip_codegen("java-jni-generate") {
+    idl = "controller-clusters.matter"
+    generator = "java"
 
-    _idl_file = "controller-clusters.matter"
-    _output_files =
-        exec_script("${chip_root}/scripts/codegen.py",
-                    [
-                      "--generator",
-                      "java",
-                      "--log-level",
-                      "fatal",
-                      "--name-only",
-                      rebase_path("controller-clusters.matter", root_build_dir),
-                    ],
-                    "list lines",
-                    [ "controller-clusters.matter" ])
-
-    args = [
-      "--generator",
-      "java",
-      "--output-dir",
-      rebase_path(target_gen_dir, root_build_dir),
-      rebase_path(_idl_file, root_build_dir),
+    outputs = [
+      "jni/IdentifyClient-ReadImpl.cpp",
+      "jni/IdentifyClient-InvokeSubscribeImpl.cpp",
+      "jni/GroupsClient-ReadImpl.cpp",
```

Comment:
Really unfortunate that we must know this in advance. In fact, it makes it so that possibly it's not feasible to include custom cluster definitions for Java without forking build files

Reply:
Not sure how feesible this is, but maybe we can add another argument to `chip_codegen` that allows for output files matching to be either strict or generated must be a superset of provided output :man_shrugging:

Reply:
I got this request from @mspang to do this. It is possible at runtime, Michael said that based on previous experience that is a bad idea.

Reply:
As a note: we *do* have a argument in chip_codegen to list all generated files, previous script for java was doing this, but the dynamic-bridge was not using it (and as such the output list was too short).

Reply:
Asked around other gn-using systems, the feedback seemed to be generally `do not use exec_script` with the major reasoning being that gn will slow down if we exec_script (supposedly by a lot).

So this is annoying (double the location of changes), but it seems to make things more explicit and speeds up builds.

Reply:
We will be able as a folloupw to have a `update build` script to auto-fill these. 
It seems large project user consensus was to not use auto-fill lists in these cases.

java gets around this by creating jar files (single known outputs) out of mixed compile results. We cannot generate  single files, so we will need to support the split and manual naming.